### PR TITLE
Windows support Review Stage

### PR DESCRIPTION
**Pull request by johnwparent**
Staging PR to address reviews on #27021. Review changes requested or discussed on the support branch should be made here to maintain a consistent status for other reviewers of the support branch CI.

### CODE REVIEW

File: lib/spack/spack/cmd/make_installer.py [hunk 1]
```diff
@@ -97,13 +97,13 @@ def make_installer(parser, args):
...
 
         try:
-            subprocess.check_call(
-                ('"%s" -S "%s" -B "%s" -DSPACK_VERSION=%s '
-                 '-DSPACK_SOURCE="%s" -DSPACK_LICENSE="%s" '
-                 '-DSPACK_LOGO="%s" -DSPACK_GIT_VERBOSITY="%s"')
-                % (cmake_path, source_dir, output_dir, spack_version, spack_source,
-                   spack_license, spack_logo, git_verbosity),
-                shell=True)
+            spack.util.executable.Executable(cmake_path)(
```
Comment (johnwparent) [inline lib/spack/spack/cmd/make_installer.py:99]: This change can be ignored for the moment. Seem to have chopped off the commit where I fix this during rebase.

File: lib/spack/spack/operating_systems/windows_os.py [hunk 2]
```diff
@@ -65,6 +67,8 @@ class WindowsOs(OperatingSystem):
         compiler_search_paths = comp_search_paths
 
     def __init__(self):
+        if platform.release() != '10':
+            raise SpackError("Spack is not supported on Windows versions older than 10")
```
Comment (scheibelp) [inline lib/spack/spack/operating_systems/windows_os.py:71]: There's also Windows 11, so this won't work on that either I assume

File: lib/spack/spack/detection/common.py [hunk 3]
```diff
@@ -196,7 +196,10 @@ def find_win32_additional_install_paths():
     windows_search_ext.extend([os.environ[key] for key
                               in os.environ.keys() if
                               add_path(key)])
+    # Add search path for default Chocolately (https://github.com/chocolatey/choco)
+    # default install directory
```
Comment (scheibelp) [inline lib/spack/spack/detection/common.py]: (minor) this says "default Chocolately default install directory"

* "default" appears twice
* I think it's called "Chocolatey"

As a side note, is this expected to be a good way to get CMake/Ninja? A short note in `getting_started` would be useful in that case.

**Review by scheibelp — APPROVED** [review]
There's a couple of minor issues, but once those are resolved this looks good to merge (and I'm generally lax about PRs on PRs).

File: lib/spack/spack/detection/common.py (hunk 3 above)
Comment (johnwparent) [reply lib/spack/spack/detection/common.py]: This is not the expected way to get Ninja or CMake (I am not aware of anyone who really uses chocolatey, just trying to cover our bases w.r.t. external find)

File: lib/spack/spack/operating_systems/windows_os.py (hunk 2 above)
Comment (johnwparent) [reply lib/spack/spack/operating_systems/windows_os.py:71]: Spack is entirely untested on Windows 11, although Windows is pretty good about backwards compatibility. I can make this a softer check and just leave a note in the getting started docs. Thoughts?

Comment (scheibelp) [reply lib/spack/spack/operating_systems/windows_os.py:71]: That sounds good: In that case this check should probably be something like `if Version(platform.release()) < Version('10'):`

Comment (johnwparent) [reply lib/spack/spack/operating_systems/windows_os.py:71]: I didn't realize the version class could be used like that. Thanks for the suggestion!

File: lib/spack/spack/detection/common.py (hunk 3 above)
Comment (scheibelp) [reply lib/spack/spack/detection/common.py]: Got it, in that case I think just updating the comment here is good.

File: lib/spack/docs/getting_started.rst [hunk 4]
```diff
@@ -1705,6 +1716,15 @@ packages known to work on Windows:
 * netlib-lapack (requires Intel Fortran)
 * openssl
 * zlib
+* perl
```
Comment (BetsyMcPhail) [inline lib/spack/docs/getting_started.rst:1726]: Now that we're starting to get a longer list, alphabetization would be helpful

**Pull request merged**